scripts/pi-hole/php/savesettings.php

```diff
@@ -53,7 +53,15 @@ function validDomainWildcard($domain_name)
 function validMAC($mac_addr)
 {
   // Accepted input format: 00:01:02:1A:5F:FF (characters may be lower case)
-  return (preg_match('/([a-fA-F0-9]{2}[:]?){6}/', $mac_addr) == 1);
+  return !filter_var($mac_addr, FILTER_VALIDATE_MAC) === false;
+}
+
+function formatMAC($mac_addr)
+{
+	preg_match("/([0-9a-fA-F]{2}[:]){5}([0-9a-fA-F]{2})/", $mac_addr, $matches);
+	if(count($matches) > 0)
```

Comment:
This would also validate
```
aa:aa:aa:aa:aa:aabb:bb:bb:bb:bb:bb
```
even if it would only take
```
aa:aa:aa:aa:aa:aa
```
due to `return $matches[0];`.

```suggestion
	if(count($matches) === 1)
```

This is not really a bug, just pointing this out.